platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/api/ConfigurationDefaultsIntegrationTest.groovy

```diff
@@ -241,6 +245,78 @@ project.status = 'foo'
         }
     }
 
+    def "defaultDependencies deprecations are properly attributed to source plugin"() {
+        BuildOperationsFixture buildOps = new BuildOperationsFixture(executer, temporaryFolder)
+
+        settingsFile << """
+            includeBuild("plugin")
+        """
+
+        file("plugin/build.gradle") << """
+            plugins {
+                id("java-gradle-plugin")
+            }
+
+            gradlePlugin {
+                plugins {
+                    transformPlugin {
+                        id = "com.example.plugin"
+                        implementationClass = "com.example.ExamplePlugin"
+                    }
+                }
+            }
+        """
+
+        file("plugin/src/main/java/com/example/ExamplePlugin.java") << """
+            package com.example;
+
+            import org.gradle.api.Plugin;
+            import org.gradle.api.Project;
+            import ${DeprecationLogger.name};
+            import ${DependencyScopeConfiguration.name};
+
+            public class ExamplePlugin implements Plugin<Project> {
+                public void apply(Project project) {
+                    project.getConfigurations().withType(DependencyScopeConfiguration.class).configureEach(conf -> {
+                        conf.defaultDependencies(deps -> {
+                            DeprecationLogger.deprecate("foo")
+                                .willBecomeAnErrorInGradle10()
+                                .undocumented()
+                                .nagUser();
+                        });
+                    });
+                }
+            }
+        """
+
+        buildFile.text = """
+            plugins {
+                id("com.example.plugin")
+            }
+
+            configurations {
+                dependencyScope("deps")
+            }
+
+            // triggers `defaultDependencies`
+            configurations.deps.incoming.dependencies
```

Comment:
💅 This is CC safe because it's not actually resolving the deps at configuration time, just triggering the defaults.  It's still a little odd to look at, it might be worth doing within a task just to be more representative of when this would typically happen.